src/adams_methods.jl

```diff
@@ -0,0 +1,655 @@
+            yerr_km1 = h*(g[steporder] - g[steporder-1])*ϕ[3,steporder]
+            err_km1 = stepsize_hw92!(h, tdir, y[n], y_np1, yerr_km1, steporder,
+                            timeout_VS,dof, abstol, reltol, maxstep, norm)[1]
+
+            yerr_k = h*(g[steporder+1] - g[steporder])*ϕ[3,steporder+1]
+            err_k = stepsize_hw92!(h, tdir, y[n], y_np1, yerr_k, steporder,
+                            timeout_VS, dof, abstol, reltol, maxstep, norm)[1]
+
+            # Decrement steporder if LE_{k} and LE_{k-1} < LE_{k+1}
+            if max(err_km1,err_k) <= err_kp1
+                steporder = max(order,steporder - 1)
+
+            elseif consec_const_steps >= steporder && timeout_VO==0 && n>steporder + 3
+                # Given enough consecutive steps taken at a constant step size
+                # we compute truncation error for order implicit k+2
+                ϕ_and_ϕstar_coefs!(b, ϕ, ϕstar,dy[n],t,t_np1,n,steporder+2)
+                ϕ_coefs!(ϕ, ϕstar,dy_np1,steporder+3)
+
+                yerr_kp2 = h*γ_star(steporder + 3)*ϕ[3,steporder+3]
+                err_kp2 = stepsize_hw92!(h,tdir,y[n],y_np1,yerr_kp2,steporder,
+                            timeout_VS, dof, abstol, reltol, maxstep, norm)[1]
+
+                # Increment steporder if LE_{k+2} < LE_{k+1} and max(LE_{k},
+                # LE_{k-1}) > LE_{k+1}
+                if err_kp2 < err_kp1
+                    steporder = min(max_order,steporder + 1)
+                    timeout_VO = timeout_VO_const # reset timout_VO
+                end
+            end
+            # Keep timeout_VO 0, or decrement
+            timeout_VO= timeout_VO==0? 0 : timeout_VO-1
+        end # end of adaptive order selection
+
+        #--#####################################################################
+        # Prepare for next step
+        #--#####################################################################
+
+        # Shift data from this step, from the "current step" position to the
+        # "previous step" position for ϕstar and ϕ
+        # TODO: implement with DataStructures.CircularBuffer
+        ϕstar[1,1:min(max(n-1,1),current_steporder+1)] =
+                                ϕstar[2,1:min(max(n-1,1),current_steporder+1)]
+        ϕ[1,1:min(max(n-1,1),current_steporder+1)] =
+                                ϕ[2,1:min(max(n-1,1),current_steporder+1)]
+
+        # Update stepsize pointers
+        prev_h = h
+        h = next_h
+        # Increment our step
+        n= n+1
+    end #while loop over steps
+
+    if points == :steps
+        return vcat(t), y
+    else
+        return vcat(tout), yout
+    end
+end
+
+###############################################################################
+#
+# HELPER FUNCTIONS and TABLEAUS FOR ADAMS SOLVERS
+#
+###############################################################################
+##Adams Bashforth Coefficients for Explicit Method
+const ms_coefficients4 = Float64[1        0        0         0
```

Comment:
These shouldn't be `Float64`.  This will result in a loss of accuracy when using something like `BigFloat`